packages/auth-nextjs/src/shared.ts

```diff
@@ -560,9 +564,16 @@ export abstract class NextAuth extends NextAuthHelpers {
               this.options.pkceVerifierCookieName,
             )?.value;
             if (!verifier) {
+              // End user verified email from a different user agent than
+              // sign-up. This is fine, but the application will need to detect
+              // this and inform the end user that they will need to initiate a
+              // new sign up attempt to complete the flow.
               return onBuiltinUICallback(
                 {
-                  error: new PKCEError("no pkce verifier cookie found"),
+                  error: null,
+                  tokenData: null,
+                  provider: null,
+                  isSignUp: false,
```

Comment:
Actually, this probably should be `true`, as email verification is part of the sign up flow.

Reply:
Hmm, actually, I don't think this works as-is: it's impossible to tell the difference between a sign in attempt from an unverified email vs. a successful verification without the PKCE verifier since they have the same shape.

Let me do a little design work here. I think we might need to introduce a discriminant into this callback payload type to make it less ambiguous for the consumer.